pkg/plugins/models.go

```diff
@@ -334,6 +334,17 @@ type Permission struct {
 	Scope  string `json:"scope"`
 }
 
+// ActionSetRegistration stores a actionset and its actions
+type ActionSetRegistration struct {
+	ActionSet ActionSet `json:"actionset"`
+}
+
+// ActionSet is the model for ActionSet in RBAC.
+type ActionSet struct {
```

Comment:
I'm thinking about management operations related to action sets, and how that may impact updates or removals of plugin actionsets later.

Updating the actions of an existing actionset is straightfoward, since you have the name of the actionset to key off of. However changing the `action` is harder. At first I considered the `action` to be the unique identifier for the action set, but I'm worried that this isn't enough. If you want to rename an action set for any reason, there wouldn't be a way to do it without creating a new one, migrating all permissions already assigned to the new one (and those created in the future for some period of time), then eventually removing the old actionset. Otherwise, if you renamed one by doing a delete and create under the new name, you lose access already granted by the previous actionset.

As far as I know the only way around this is for the plugin author to specify a UID for each actionset. This is a bit of extra work but it could be a string derived from existing values on the plugin like it's ID.

Curious if you've already considered some of the day 2 operations that plugin authors might make with action sets?

Reply:
@gamab i think me and gabriel talked about this. please correct me if i am wrong gabriel. We cannot hinder plugin authors too much about how to setup their own actionsets. We should absolutely support renaming a actionset.

The solutions might be for plugin authors, in order of ease:
- recommend to only build new actionsets and migrate to new once when permissions has been reassigned
- modify an existing actionset

> If you want to rename an action set for any reason, there wouldn't be a way to do it without creating a new one, migrating all permissions already assigned to the new one (and those created in the future for some period of time), then eventually removing the old actionset. 

honestly i think this is a viable strategy in my opinon. We do the same with some columns in databases. This might be the recommended approach from us and the same for us as well